app/controllers/application_controller.rb

```diff
@@ -9,11 +9,11 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
   def is_admin?
-    @is_admin = logged_in? ? current_user.admin : false
+    @is_admin ||= logged_in? ? current_user.admin : false
```

Comment:
Is it really safe to add an ||= here? I am not completely clear of the functionality, but this may create unintended consequences, e.g. if logged_in? or  current_user.admin is changed, then is_admin will not become false.

Reply:
My primary concern was the state of the variable upon logging out and such that might change the status of @is_admin and @is_teacher, but I asked Michael and he said it should be safe since it should be reset upon logout or something. Not sure if your concern is elsewhere, might be best to ask Michael?

We can remove the @is_admin and @is_teacher, but there's some Navbar stuff that is completely dependent on those variables. There might some other stuff as well, since that variable was part of the code before Iteration 1.